src/modules/universal-discovery/components/tab-content/bookmarks.panel.component.js

```diff
@@ -0,0 +1,90 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+import TabContentPanelComponent from './tab.content.panel.component';
+import BookmarksComponent from '../bookmarks/bookmarks.component';
+
+import './css/bookmarks.panel.component.css';
+
+export default class BookmarksPanelComponent extends Component {
+    constructor() {
+        super();
+
+        this._refBookmarksComponent = null;
+    }
+
+    onBookmarkRemoved(item) {
+        if (this._refBookmarksComponent) {
+            this._refBookmarksComponent.onBookmarkRemoved(item);
+        }
+    }
+
+    onBookmarkAdded(item) {
+        if (this._refBookmarksComponent) {
+            this._refBookmarksComponent.onBookmarkAdded(item);
+        }
+    }
+
+    getBookmarksAttrs() {
+        const {
+            multiple,
+            findContentBySearchQuery,
+            onItemSelect,
+            maxHeight,
+            contentTypesMap,
+            bookmarksPerPage,
+            labels,
+            restInfo
+        } = this.props;
+
+        return {
+            multiple,
+            findContentBySearchQuery,
+            onItemSelect,
+            maxHeight,
+            contentTypesMap,
+            bookmarksPerPage,
+            labels,
+            restInfo
+        };
+    }
+
+    getWrapperAttrs() {
+        const { isVisible } = this.props;
+
+        return {
+            hidden: !isVisible
+        };
+    }
+
+    render() {
+        const wrapperAttrs = this.getWrapperAttrs();
+        const bookmarksAttrs = this.getBookmarksAttrs();
+
+        return (
+            <div className='c-bookmarks-panel' {...wrapperAttrs}>
+                <TabContentPanelComponent {...this.props}>
+                    <BookmarksComponent
+                        ref={(ref) => this._refBookmarksComponent = ref}
```

Comment:
Please create a function with this assignment, as it's done in other places.

Reply:
In other places in this repository, it is done in that way. Only in one place, in this PR, it is done via a separate function, in `UniversalDiscoveryModule`; `assignBookmarksPanelComponentRef`,  where I have done that to improve code readability.

Reply:
Sorry, we are switching to this approach (probably more in PB). Please use it.

Reply:
I think @dew326 meant for Page Builder.